src/components/containers/seo.js

```diff
@@ -16,20 +17,39 @@ const SEO = ({ description, meta, title, no_index }) => {
                         author
                     }
                 }
+
+                allSitePage {
+                    nodes {
+                        path
+                    }
+                }
             }
         `,
     )
-    const metaDescription = description || site.siteMetadata.description
-    const { locale: lang } = React.useContext(LocaleContext)
 
+    const metaDescription = description || queries.site.siteMetadata.description
```

Comment:
@sara-fs SEO component will be rendered in every page, by doing this, we are looping on all pages in every page and pushing it in every page. you can use `path` in the query `site` instead, and push it directly